plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -973,9 +979,40 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
+            String stsEndpointOverwrite = conf.get(S3_STS_ENDPOINT);
+            String stsRegionOverwrite = conf.get(S3_STS_REGION);
+
+            AWSSecurityTokenServiceClientBuilder stsClientBuilder =
+                    AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(provider);
```

Comment:
You can inline the first two lines
```suggestion
            AWSSecurityTokenServiceClientBuilder stsClientBuilder = AWSSecurityTokenServiceClientBuilder.standard()
                    .withCredentials(provider);
```

Reply:
I'd also move this down to right before it's used